sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
I'm fine with this as a transitionary measure, but going forward we definitely need a better solution than this (even inside core). What is the story going forward for this?

Personally, I believe the solution in #2797 (allow multiple loading specs) is the way to go forward. But I'm not sure you like it that much. In practical terms, it would simply move `CommonJSName` into the `.sjsir` and everything would be gone (up to the build author to make sure their JSEnv and module mode are in sync).